BWT/Program.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System.Linq.Expressions;
+using System.Numerics;
+using System.Reflection.Metadata;
+
+int[] SuffixArrayBuild(string str)
+{
+    int[] shifts = new int[str.Length];
+    int[] countingSortHelper = new int[str.Length];
+    for (int i = 0; i < str.Length; ++i)
+    {
+        shifts[i] = i;
+    }
+    Array.Sort(shifts, (int a, int b) => { if (str[a] < str[b]) return -1; if (str[a] > str[b]) return 1; return 0; });
```

Comment:
Еще не нужно писать функции в одну строчку, читаемость сильно падает
```suggestion
    Array.Sort(shifts, (int a, int b) => 
    { 
        if (str[a] < str[b]) 
            return -1; 
        return str[a] < str[b] ? 1 : 0
    });
```